### PR TITLE
Uptime altered to use templates

### PR DESCRIPTION
**Pull request by Sloff**
This updates the uptime module to use the new template system as part of #1163 

**Before**
before
before2

**After**
after
after2

### CODE REVIEW

File: lib/DDG/Goodie/Uptime.pm [hunk 1]
```diff
@@ -19,7 +19,8 @@ topics "computing";
 code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/Uptime/Uptime.pm";
 attribution github => ["YouriAckx", "Youri Ackx"],
             web => ["http://ackx.net/", "Youri Ackx"],
-            twitter => ["YouriAckx", "Youri Ackx"];
+            twitter => ["YouriAckx", "Youri Ackx"],
```
Comment (mintsoft) [inline lib/DDG/Goodie/Uptime.pm]: I think you meant to put yourself here :)

Comment (Sloff) [inline lib/DDG/Goodie/Uptime.pm]: Hi @mintsoft, I added myself just underneath, ```github => ["https://github.com/Sloff/", "Sloff"];```

File: lib/DDG/Goodie/Uptime.pm [hunk 2]
```diff
@@ -63,21 +64,45 @@ sub format_text {
...
+                group => "text",
+                moreAt => 0
+            }
+        }
+    }
+    
+    return structured_answer => {
+        id => "uptime",
+        name => "Answer",
+        description => "Implied downtimes for $uptime_percentage uptime",
```
Comment (moollaza) [inline lib/DDG/Goodie/Uptime.pm]: This isn't a valid property for the `data` hash, you can remove it.

File: lib/DDG/Goodie/Uptime.pm [hunk 3]
```diff
@@ -63,21 +64,45 @@ sub format_text {
...
-        return $html;
+        return structured_answer => {
+            id => "uptime",
+            name => "Answer",
+            data => {
+                title => "No downtime or less than a second during a year",
+                subtitle => "Implied downtimes for $uptime_percentage uptime"
+            },
+            templates => {
+                group => "text",
```
Comment (moollaza) [inline lib/DDG/Goodie/Uptime.pm]: The `list` template also supports a `title` and `subtitle`. If you switch to that, you could possibly reduce some of the duplication here and rather add the `record_data` and `record_keys` to the hash when necessary?

**Comment by moollaza** [comment]
@Sloff thanks a lot for this! Overall this is looking pretty great.

There's just a couple things that need to be addressed and then this should be good for merge!

**Comment by Sloff** [comment]
Thanks @moollaza, I'll implement those changes ASAP!

**Comment by Sloff** [comment]
Hi @moollaza, changes made as suggested

File: lib/DDG/Goodie/Uptime.pm (hunk 1 above)
Comment (mintsoft) [inline lib/DDG/Goodie/Uptime.pm]: Doh! I should learn to read!

File: lib/DDG/Goodie/Uptime.pm [hunk 4]
```diff
@@ -102,10 +126,10 @@ handle remainder => sub {
     return unless $perl_number >= 0 && $perl_number < 100;
     
     my ($year, $month, $day) = compute_durations($perl_number / 100);
-    my $text = format_text($clean_query, $year, $month, $day);
-    my $html = format_html($clean_query, $year, $month, $day);
+    my $plaintext = format_text($clean_query, $year, $month, $day);
+    my @structured_answer = format_answer($clean_query, $year, $month, $day);
     
-	return $text, html => $html;
+	return $plaintext, @structured_answer;
```
Comment (mintsoft) [inline lib/DDG/Goodie/Uptime.pm]: Can you switch this hard-tab out for 4 spaces?

File: t/Uptime.t [hunk 5]
```diff
@@ -8,47 +8,133 @@ use DDG::Test::Goodie;
...
-    'uptime 99.9999%' => test_zci(qr/.*/, html=>qr/.*Annually: 31 seconds.*/),
-    'uptime 99.9999%' => test_zci(qr/.*/, html=>qr/.*Monthly: 2 seconds.*/),
-    'uptime 99.9999%' => test_zci(qr/.*/, html=>qr/.*Daily: less than one second.*/),
-    'uptime 99.99999%' => test_zci(qr/.*/, html=>qr/.*Annually: 3 seconds.*/),
-    'uptime 99.99999%' => test_zci(qr/.*/, html=>qr/.*Monthly: less than one second.*/),
-    'uptime 99.99999%' => test_zci(qr/.*/, html=>qr/.*Daily: less than one second.*/),
-
-    # Lower limit
-    'uptime 0%' => test_zci(qr/.*/, html=>qr/.*0% uptime.*/),
-    'uptime 000%' => test_zci(qr/.*/, html=>qr/.*000% uptime.*/),
```
Comment (mintsoft) [inline t/Uptime.t]: We appear to have lost test coverage on `uptime 0%` here

**Comment by Sloff** [comment]
Hi @mintsoft, changes made as suggested

File: lib/DDG/Goodie/Uptime.pm (hunk 1 above)
Comment (Sloff) [inline lib/DDG/Goodie/Uptime.pm]: Haha, no problem

**Comment by mintsoft** [comment]
Thanks for the quick turnaround @Sloff

**Pull request merged**